Apps/System/music.c

```diff
@@ -0,0 +1,461 @@
+/* music.c
+ *
+ * Music player
+ *
+ * RebbleOS
+ * 
+ * Author: Chris Multhaupt <chris.multhaupt@gmail.com>.
+ */
+
+#include "rebbleos.h"
+#include "music.h"
+#include "property_animation.h"
+
+#define RECORD_CENTER_X 56
+#define RECORD_CENTER_Y 57
+#define RECORD_CENTER GPoint(RECORD_CENTER_X, RECORD_CENTER_Y)
+#define RECORD_SWOOSH_DISTANCE 130
+#define RECORD_SWOOSH_SPEED 250
+#define ARM_OFFSET_X 17
+#define ARM_OFFSET_Y 77
+#define ARM_OFFSET GPoint(ARM_OFFSET_X, ARM_OFFSET_Y)
+#define ARM_OFFSET_ANGLE -31
+#define ARM_HOME_ANGLE 0
+#define ARM_START_ANGLE 18
+#define ARM_END_ANGLE 46
+#define ARM_SKIP_SPEED 200
+#define ARM_QUICK_SPEED 100
+#define SKIP_DIRECTION_PREV -1
+#define SKIP_DIRECTION_NEXT 1
+#define PROGRESS_PIXELS_MAX 105
+
+static Window *s_main_window;
+static Layer *s_main_layer;
+static ActionBarLayer *s_action_bar;
+
+static n_GPath *s_arm_base_path_ptr;
+static n_GPath *s_arm_1_path_ptr;
+static n_GPath *s_arm_2_path_ptr;
+static n_GPath *s_arm_head_path_ptr;
+
+static const n_GPathInfo ARM_BASE_PATH_INFO = {
+    .num_points = 2,
+    .points = (n_GPoint[]) {{0, -3}, {0, 3}}
+};
+static const n_GPathInfo ARM_1_PATH_INFO = {
+    .num_points = 2,
+    .points = (n_GPoint[]) {{0, -3}, {0, -3-19}}
+};
+static const n_GPathInfo ARM_2_PATH_INFO = {
+    .num_points = 2,
+    .points = (n_GPoint[]) {{0, -3-19}, {19, -3-19-15}}
+};
+static const n_GPathInfo ARM_HEAD_PATH_INFO = {
+    .num_points = 2,
+    .points = (n_GPoint[]) {{19, -3-19-14}, {19+9, -3-19-15-6}}
+};
+
+static Animation *s_animation_record_ptr;
+static Animation *s_animation_arm_ptr;
+static bool s_animating_disk_change;
+static bool s_animating_arm_change;
+static bool s_is_paused;
+static bool s_was_paused;
+static char *s_artist;
+static char *s_track;
+static GBitmap *s_up_bitmap;
+static GBitmap *s_down_bitmap;
+static GBitmap *s_select_bitmap;
+static GColor s_curr_disk_color;
+static GColor s_next_disk_color;
+static GPoint s_curr_disk_pos;
+static GPoint s_next_disk_pos;
```

Comment:
One thing to think about with all these static structs: since this is linked into ROM, these effectively exist in the System heap.  Maybe encapsulating them all in a struct (ugh...), which we `app_calloc` on music app init, which would put them in the app heap?

Reply:
This proposal is roughly what we did at pebble for all system apps. We had a little wrapper around it.